ktor-features/ktor-graphql/src/io/ktor/graphql/renderGraphiQL.kt

```diff
@@ -0,0 +1,158 @@
+package io.ktor.graphql
+
+const val GRAPHIQL_VERSION = "0.11.11"
+const val JS_UNDEFINED = "undefined"
+
+internal fun renderGraphiQL(data: Map<String, Any?>?, request: GraphQLRequest): String {
+
+    val queryString = request.query
+
+    val dataJSON = convertToJSONString(data)
+    val variablesJSON = convertToJSONString(request.variables)
+
+    val operationName = request.operationName
+
+    val value =  """<!--
+    The request to this GraphQL server provided the header "Accept: text/html"
+    and as a result has been presented GraphiQL - an in-browser IDE for
+        exploring GraphQL.
+    If you wish to receive JSON, provide the header "Accept: application/json" or
+    add "&raw" to the end of the URL within a browser.
+    -->
+    <!DOCTYPE html>
+    <html>
+    <head>
+    <meta charset="utf-8" />
+    <title>GraphiQL</title>
+    <meta name="robots" content="noindex" />
+    <meta name="referrer" content="origin" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <style>
+            body {
+                margin: 0;
+                overflow: hidden;
+            }
+    #graphiql {
+        height: 100vh;
+    }
+    </style>
+    <link href="//cdn.jsdelivr.net/npm/graphiql@${GRAPHIQL_VERSION}/graphiql.css" rel="stylesheet" />
+    <script src="//cdn.jsdelivr.net/es6-promise/4.0.5/es6-promise.auto.min.js"></script>
+    <script src="//cdn.jsdelivr.net/fetch/0.9.0/fetch.min.js"></script>
+    <script src="//cdn.jsdelivr.net/react/15.4.2/react.min.js"></script>
+    <script src="//cdn.jsdelivr.net/react/15.4.2/react-dom.min.js"></script>
+    <script src="//cdn.jsdelivr.net/npm/graphiql@${GRAPHIQL_VERSION}/graphiql.min.js"></script>
```

Comment:
Can you include the subresource integrity check hashes? Preferably something SHA-256 or higher.

https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity